pkg/controller/lifecyclehandler/handler.go

```diff
@@ -377,13 +388,13 @@ func setObservedGeneration(resource *k8s.Resource, observedGeneration int64) {
 	resource.Status["observedGeneration"] = observedGeneration
 }
 
-func (r *LifecycleHandler) recordEvent(resource *k8s.Resource, eventtype, reason, message string) error {
+func (r *LifecycleHandler) recordEvent(resource *k8s.Resource, eventtype, reason, message string) {
 	u, err := resource.MarshalAsUnstructured()
 	if err != nil {
-		return err
+		// todo acpana log err
```

Comment:
Yeah, a sternly worded warning is probably fine here.  I don't love the fact that k8s.Resource is not a wrapper around unstructured.Unstructured or client.Object anyway.